framework/src/utils/PetscSupport.C

```diff
@@ -533,18 +575,42 @@ storePetscOptions(FEProblemBase & fe_problem, const InputParameters & params)
   Moose::PetscSupport::PetscOptions & po = fe_problem.getPetscOptions();
 
   // First process the single petsc options/flags
-  processPetscFlags(petsc_options, po);
+  processPetscFlags(petsc_options, prefix, param_object, po);
 
   // Then process the option-value pairs
-  processPetscPairs(petsc_pair_options, fe_problem.mesh().dimension(), po);
+  processPetscPairs(petsc_pair_options, fe_problem.mesh().dimension(), prefix, param_object, po);
+}
+
+#define checkPrefix(prefix)                                                                        \
+  mooseAssert(prefix[0] == '-',                                                                    \
+              "Leading prefix character must be a '-'. Current prefix is '" << prefix << "'");     \
+  mooseAssert((prefix.size() == 1) || (prefix.back() == '_'),                                      \
+              "Terminating prefix character must be a '_'. Current prefix is '" << prefix << "'")
```

Comment:
do we have this check for user-provided CLI petsc parameters? To make sure they included a prefix?